tap_sftp/singer_encodings/json_schema.py

```diff
@@ -38,10 +38,11 @@ def sample_file(conn, table_spec, f, sample_rate, max_records, config):
         file_handle = conn.get_file_handle(f)
 
     # Add file_name to opts and flag infer_compression to support gzipped files
-    opts = {'key_properties': table_spec['key_properties'],
+    opts = {'key_properties': table_spec.get('key_properties', ['_sdc_source_file', '_sdc_source_lineno']),
```

Comment:
@radbrt it looks like this is new behavior of adding default key properties if they arent specified. Is that right? I dont really the implications of this. Is there a chance of users who currently dont have key properties set, getting them added after the PR and upgrading their version? I'm not sure it would matter but want to make sure we dont introduce any breaking changes by accident.

Reply:
Agreed, My thought was that I would provide a default in case nothing was specified, but this would override if the tap was configured with an empty list. And that would be a breaking change, which we don't want. Removing.

Reply:
@radbrt sorry for the slow response. If you "@" me with new comments I'll get a notification so I dont miss anything :smile:!

>My thought was that I would provide a default in case nothing was specified, but this would override if the tap was configured with an empty list. And that would be a breaking change

I actually think it would respect an empty list because the `get('key_properties'` so only null key properties would have a change. The more I think about it I think I'm actually in favor of making that the default, especially because this is a new repo and we dont have any releases yet so most likely users are using the bonobos variant still or are pinning commit hashes.

What do you think? If you agree, would you mind adding a note in the readme explaining it and also add `sanitize_header` as one of the options available for the tables entries.

FWIW I think this is doing the same thing so whatever we decide we should make sure its consistent across the two https://github.com/MeltanoLabs/tap-sftp/pull/11/files#diff-cb5a8465169f4b210c9601bbc6ba2988f90598e197fd637dc378f0ca20e74cf5L63.

Reply:
Thanks @pnadolny13 

I tried it out just to be sure, and the `get` method does seem to respect an empty list. So basically:

- Existing installations must have configured a key, otherwise it would have failed. So there are no functioning existing installations without a `key_properties` config key set.
- If we add a default in the `get` property, it will respect anything configured (including empty lists), except if no `key_properties` are set, in which case it wouldn't be a functioning installation. So adding a default would not impact any currently functioning installation.

It does seem fairly safe to add a default here.

I keep forgetting that the parameters are passed through in two different files, so thanks for reminding me I need to change it two places.

In summary, it seems we agree about the following two changes:
- revert to a default key_properties, and follow through by marking it as optional
- make sure we pass those defaults on in all places that propagates config
- document the sanitize_header config key.

As a final question: I did not put much thought into the `sanitize_header` key name, and I don't know if it is the best name. I have toyed around with `clean_colnames` as an alternative, but that might be more ambigious. Thoughts?

Reply:
@radbrt yes that summary sounds correct to me! Personally I'm fine with `sanitize_header` for naming. Once we have an entry in the README and MeltanoHub it should be clear what it does.

Reply:
@pnadolny13 I think the newest updates should do it. Tests passed, have tested manually with some different configurations to make sure an empty key_property key won't be overwritten, but that when the config key isn't specified the default kicks in.